CMakeLists.txt

```diff
@@ -1,9 +1,8 @@
-cmake_minimum_required(VERSION 3.5.0 FATAL_ERROR)
-
-enable_language(Fortran)
-
-project(stdlib)
-
+cmake_minimum_required(VERSION 3.5.0)
+project(stdlib Fortran)
 enable_testing()
 
+# this avoids stdlib and projects using stdlib from having to introspect stdlib's directory structure
+set(CMAKE_Fortran_MODULE_DIRECTORY ${CMAKE_BINARY_DIR})
```

Comment:
@scivision This is a pretty big hammer. For now, I'm fine with this, but it comes with its own complexities and issues. Eventually it would be nice to install a package-config (`.pc`) file and a [CMake package configuration file](https://cmake.org/cmake/help/latest/manual/cmake-packages.7.html#creating-a-package-configuration-file) that could help consumers use the software if it's installed somewhere.

The drawbacks of this design choice are that:

1. All client code will assume modules will always be in the same place: at the root of the build directory
2. Modules with the same name will clobber each other (as was the case for the bug I found and fixed in your CMake with submodules repository)
3. Any testing & infrastructure modules that we may want to keep private to this project now get mixed in with modules intended to be consumed by client code.

Overall I think this move makes sense right now, but that a more nuanced approach is warranted in the future.

Reply:
yes more typically I set Fortran_Module_Directory as a property on each and every target, perhaps with foreach() loop if lots of targets. Just trying to level up from `include_directories()` until things get further along.

Reply:
What problems does setting `CMAKE_Fortran_MODULE_DIRECTORY` fix? I see the drawbacks, that @zbeekman described, but I do not see the advantages.